platform/website/src/lib/utils.ts

```diff
@@ -93,6 +93,19 @@ function formatSmallNumber(number: number): string {
 	return `${number}`;
 }
 
+// This is a groupBy function for arrays
+// The official function sadly has bad browser support: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy
+export function groupBy<T>(a: T[], f: (x: T) => number) {
+	return a.reduce((res: { [key: string]: T[] }, x) => {
+		const key = f(x);
+		if (!res[key]) {
+			res[key] = [];
+		}
+		res[key].push(x);
+		return res;
+	}, {});
+}
+
```

Comment:
Do you want to add it to the object prototype or not?

Reply:
I prefer not to since it can get confusing, especially because groupBy isn't the name in all browsers: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy#browser_compatibility